gossip/c_block_callbacks.go

```diff
@@ -178,11 +178,20 @@ func consensusCallbackBeginBlockFn(
 					}
 				}
 
-				var block = &inter.Block{
-					Time:    blockCtx.Time,
-					Atropos: cBlock.Atropos,
-					Events:  hash.Events(confirmedEvents),
-				}
+				prevRandao := computePrevRandao(confirmedEvents)
+
+				number := uint64(blockCtx.Idx)
+				lastBlockHeader := evmStateReader.GetHeaderByNumber(number - 1)
+
+				maxBlockGas := es.Rules.Blocks.MaxBlockGas
+
+				blockBuilder := inter.NewBlockBuilder().
```

Comment:
Ideally, the builder should be created inside the blockFn lambda, to avoid state pollution in case this function is patched in the future.